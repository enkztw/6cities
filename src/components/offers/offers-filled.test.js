import React from 'react';
import {mount} from 'enzyme';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {BrowserRouter as Router} from 'react-router-dom';

import OffersFilled from './offers-filled.jsx';

const offers = [
  {
    id: 1,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-02.jpg`, `img/apartment-03.jpg`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 1.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 100,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 2,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    previewImage: `img/apartment-02.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-03.jpg`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 2.8,
    type: `room`,
    bedrooms: 2,
    maxAdults: 4,
    price: 140,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198,
      zoom: 8
    }
  },
  {
    id: 3,
    city: {
      name: `Paris`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    previewImage: `img/apartment-02.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-03.jpg`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `room`,
    bedrooms: 2,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198,
      zoom: 8
    }
  }
];

const mockStore = configureStore([]);

it(`<OffersFilled /> should render offers`, () => {
  const div = global.document.createElement(`div`);
  global.document.body.appendChild(div);

  const store = mockStore({
    offers: {
      sorting: `id`,
      data: offers
    }
  });

  const tree = mount(
      <Provider store={store}>
        <Router>
          <OffersFilled offers={offers} />
        </Router>
      </Provider>, {attachTo: div});

  expect(tree.getDOMNode()).toMatchSnapshot();
});
