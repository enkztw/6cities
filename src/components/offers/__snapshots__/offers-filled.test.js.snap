// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OffersFilled /> should render offers 1`] = `
<div
  class="cities"
>
  <div
    class="cities__places-container container"
  >
    <section
      class="cities__places places"
    >
      <h2
        class="visually-hidden"
      >
        Places
      </h2>
      <b
        class="places__found"
      >
        4
         places to stay in 
      </b>
      <form
        action="#"
        class="places__sorting"
        method="get"
      >
        <span
          class="places__sorting-caption"
        >
          Sort by
        </span>
        <span
          class="places__sorting-type"
          tabindex="0"
        >
          Popular
          <svg
            class="places__sorting-arrow"
            height="4"
            width="7"
          >
            <use
              xlink:href="#icon-arrow-select"
            />
          </svg>
        </span>
        <ul
          class="places__options places__options--custom"
        >
          <li
            class="places__option places__option--active"
            tabindex="0"
          >
            Popular
          </li>
          <li
            class="places__option"
            tabindex="0"
          >
            Price: low to high
          </li>
          <li
            class="places__option"
            tabindex="0"
          >
            Price: high to low
          </li>
          <li
            class="places__option"
            tabindex="0"
          >
            Top rated first
          </li>
        </ul>
      </form>
      <div
        class="cities__places-list places__list tabs__content"
      >
        <article
          class="cities__place-card place-card"
        >
          <div
            class="cities__image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                class="place-card__image"
                height="200"
                src="/img/room.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            class="place-card__info"
          >
            <div
              class="place-card__price-wrapper"
            >
              <div
                class="place-card__price"
              >
                <b
                  class="place-card__price-value"
                >
                  €
                  110
                </b>
                <span
                  class="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                class="place-card__bookmark-button button place-card__bookmark-button--active"
                type="button"
              >
                <svg
                  class="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlink:href="#icon-bookmark"
                  />
                </svg>
                <span
                  class="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              class="place-card__rating rating"
            >
              <div
                class="place-card__stars rating__stars"
              >
                <span
                  style="width: 100%;"
                />
                <span
                  class="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              class="place-card__name"
            >
              <a
                href="/offer/1"
              >
                Beautiful & luxurious apartment at great location
              </a>
            </h2>
            <p
              class="place-card__type"
            >
              Apartment
            </p>
          </div>
        </article>
        <article
          class="cities__place-card place-card"
        >
          <div
            class="place-card__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            class="cities__image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                class="place-card__image"
                height="200"
                src="/img/apartment-01.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            class="place-card__info"
          >
            <div
              class="place-card__price-wrapper"
            >
              <div
                class="place-card__price"
              >
                <b
                  class="place-card__price-value"
                >
                  €
                  120
                </b>
                <span
                  class="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                class="place-card__bookmark-button button place-card__bookmark-button--active"
                type="button"
              >
                <svg
                  class="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlink:href="#icon-bookmark"
                  />
                </svg>
                <span
                  class="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              class="place-card__rating rating"
            >
              <div
                class="place-card__stars rating__stars"
              >
                <span
                  style="width: 80%;"
                />
                <span
                  class="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              class="place-card__name"
            >
              <a
                href="/offer/2"
              >
                Wood and stone place
              </a>
            </h2>
            <p
              class="place-card__type"
            >
              Room
            </p>
          </div>
        </article>
        <article
          class="cities__place-card place-card"
        >
          <div
            class="cities__image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                class="place-card__image"
                height="200"
                src="/img/apartment-02.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            class="place-card__info"
          >
            <div
              class="place-card__price-wrapper"
            >
              <div
                class="place-card__price"
              >
                <b
                  class="place-card__price-value"
                >
                  €
                  240
                </b>
                <span
                  class="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                class="place-card__bookmark-button button"
                type="button"
              >
                <svg
                  class="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlink:href="#icon-bookmark"
                  />
                </svg>
                <span
                  class="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              class="place-card__rating rating"
            >
              <div
                class="place-card__stars rating__stars"
              >
                <span
                  style="width: 40%;"
                />
                <span
                  class="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              class="place-card__name"
            >
              <a
                href="/offer/3"
              >
                Canal View Prinsengracht
              </a>
            </h2>
            <p
              class="place-card__type"
            >
              House
            </p>
          </div>
        </article>
        <article
          class="cities__place-card place-card"
        >
          <div
            class="place-card__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            class="cities__image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                class="place-card__image"
                height="200"
                src="/img/apartment-03.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            class="place-card__info"
          >
            <div
              class="place-card__price-wrapper"
            >
              <div
                class="place-card__price"
              >
                <b
                  class="place-card__price-value"
                >
                  €
                  50
                </b>
                <span
                  class="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                class="place-card__bookmark-button button"
                type="button"
              >
                <svg
                  class="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlink:href="#icon-bookmark"
                  />
                </svg>
                <span
                  class="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              class="place-card__rating rating"
            >
              <div
                class="place-card__stars rating__stars"
              >
                <span
                  style="width: 60%;"
                />
                <span
                  class="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              class="place-card__name"
            >
              <a
                href="/offer/4"
              >
                Nice, cozy, warm big bed apartment
              </a>
            </h2>
            <p
              class="place-card__type"
            >
              Hotel
            </p>
          </div>
        </article>
      </div>
    </section>
    <div
      class="cities__right-section"
    >
      <section
        class="cities__map map leaflet-container leaflet-grab leaflet-touch-drag"
        id="map"
        style="position: relative;"
        tabindex="0"
      >
        <div
          class="leaflet-pane leaflet-map-pane"
          style="left: 0px; top: 0px;"
        >
          <div
            class="leaflet-pane leaflet-tile-pane"
          >
            <div
              class="leaflet-layer "
              style="z-index: 1;"
            >
              <div
                class="leaflet-tile-container leaflet-zoom-animated"
                style="z-index: 18; left: 0px; top: 0px;"
              >
                <img
                  alt=""
                  class="leaflet-tile"
                  role="presentation"
                  src="https://b.basemaps.cartocdn.com/rastertiles/voyager_labels_under/12/2103/1345{@2x}.png"
                  style="width: 256px; height: 256px; left: -192px; top: -219px;"
                />
              </div>
            </div>
          </div>
          <div
            class="leaflet-pane leaflet-shadow-pane"
          />
          <div
            class="leaflet-pane leaflet-overlay-pane"
          />
          <div
            class="leaflet-pane leaflet-marker-pane"
          >
            <img
              alt=""
              class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
              src="/img/pin.svg"
              style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -136px; top: -36px; z-index: -36;"
              tabindex="0"
            />
            <img
              alt=""
              class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
              src="/img/pin.svg"
              style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -136px; top: 66px; z-index: 66;"
              tabindex="0"
            />
            <img
              alt=""
              class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
              src="/img/pin.svg"
              style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 86px; top: -36px; z-index: -36;"
              tabindex="0"
            />
            <img
              alt=""
              class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
              src="/img/pin.svg"
              style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 115px; top: 11px; z-index: 11;"
              tabindex="0"
            />
          </div>
          <div
            class="leaflet-pane leaflet-tooltip-pane"
          />
          <div
            class="leaflet-pane leaflet-popup-pane"
          />
        </div>
        <div
          class="leaflet-control-container"
        >
          <div
            class="leaflet-top leaflet-left"
          />
          <div
            class="leaflet-top leaflet-right"
          />
          <div
            class="leaflet-bottom leaflet-left"
          />
          <div
            class="leaflet-bottom leaflet-right"
          >
            <div
              class="leaflet-control-attribution leaflet-control"
            >
              <a
                href="https://leafletjs.com"
                title="A JS library for interactive maps"
              >
                Leaflet
              </a>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;
